modules/homa-lite/src/lib.rs

```diff
@@ -567,6 +589,53 @@ pub mod module {
 
 			Ok(())
 		}
+
+		/// Adjusts the AvailableStakingBalance by the given difference.
+		/// Also attempt to process queued redeem request with the new Staking Balance.
+		/// Requires `T::GovernanceOrigin`
+		///
+		/// Parameters:
+		/// - `adjustment`: The difference in amount the AvailableStakingBalance should be adjusted
+		///   by.
+		///
+		/// Weight: Weight(xcm unbond) + n * Weight(match redeem requests), where n is number of
+		/// redeem requests matched.
+		#[pallet::weight(
+			< T as Config >::WeightInfo::adjust_available_staking_balance_with_no_matches().saturating_add(
+			max_num_matches.saturating_mul(< T as Config >::WeightInfo::redeem_with_available_staking_balance())
+			)
+		)]
+		#[transactional]
+		pub fn adjust_available_staking_balance(
+			origin: OriginFor<T>,
+			by_amount: AmountOf<T>,
+			max_num_matches: u64,
```

Comment:
why u64? u32 is already more than enough

Reply:
I originally used u64 to avoid conversion to Weights, but I think you are right - it is more correct to use u32